src/Util/ActorDimensionKeeper.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <math/seadVector.h>
+
+namespace al {
+    class LiveActor;
+    class MtxConnector;
+}
+
+class In2DAreaMoveControl;
+class IUsePlayerCollision;
+
+class ActorDimensionKeeper {
+private:
+    const al::LiveActor* mLiveActor;
+    bool mIsValid = true;
+    bool mIs2D = false;
+    bool mIsIn2DArea = false;
+    bool mIsCurrently2D = false;
+    bool mIsCanChange2D = false;
+    bool mIsCanChange3D = false;
+    In2DAreaMoveControl* mIn2DAreaMoveControl = nullptr;
+
+public:
+    ActorDimensionKeeper(const al::LiveActor* actor);
+    void validate();
+    void invalidate();
+    void forceChange2DKeep();
+    void forceEndChange2DKeep();
+    void update();
+
+    bool getIs2D() const { return mIs2D; }
+    bool getIsIn2DArea() const { return mIsIn2DArea; }
+    bool getIsCurrently2D() const { return mIsCurrently2D; }
+    bool getIsCanChange2D() const { return mIsCanChange2D; }
+    bool getIsCanChange3D() const { return mIsCanChange3D; }
+};
+
+class IUseDimension {
+public:
+    virtual ActorDimensionKeeper* getActorDimensionKeeper() const = 0;
```

Comment:
```suggestion
    virtual const ActorDimensionKeeper* getActorDimensionKeeper() const = 0;
```

Reply:
This is more hope than knowledge. Please verify.

Reply:
Works. Changing it